internal/cosmetic/store/store.go

```diff
@@ -0,0 +1,67 @@
+package store
+
+import (
+	"strings"
+	"sync"
+)
+
+type RuleStore struct {
+	mu    sync.RWMutex
+	store map[string][]string
+}
+
+func NewStore() *RuleStore {
+	return &RuleStore{
+		store: make(map[string][]string),
+	}
+}
+
+func (rs *RuleStore) Add(hostnames []string, selector string) {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	if len(hostnames) == 0 {
+		rs.store["*"] = append(rs.store["*"], selector)
```

Comment:
I'd suggest using a separate variable for `rs.store["*"]`, since hashmaps require quite a lot of work to retrieve an element on every access.

Reply:
Could you help me with this one? I'm not sure if there’s a way to reduce the number of hashmap retrieval operations. We still need to read the existing selectors for `"*"` and then reassign it, which results in two operations.

The only solution I came up with is this:

```go
if len(hostnames) == 0 {
    selectors := rs.store["*"]
    selectors = append(selectors, selector)
    rs.store["*"] = selectors

    return nil
}
```

However, this still requires two hashmap operations.